sqlstats: remove sqlstats controller structs

xinhaoz opened the pull request:
### sqlstats: remove move persistedsqlstats controller
Let's remove this controller struct and just use the PersistedSQLStats provider
directly for clarity.

Epic: none
Release note: None

### sqlstats: remove sslocal controller
This struct is not very useful as it just provides an interface to reset local sql stats. This commit removes the sql stats controller defined in sslocal as we can access the local sql stats directly.

Epic: none

Release note: None

--------------

Please note only the latest 2 commits should be reviewed in this PR.

xinhaoz committed:
sqlstats: remove sslocal controller

This struct is not very useful as it just provides an interface to reset local
sql stats. This commit removes the sql stats controller defined in sslocal as we
 can access the local sql stats directly.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144417)
<!-- Reviewable:end -->

xinhaoz committed:
sqlstats: remove move persistedsqlstats controller

Let's remove this controller struct and just use the PersistedSQLStats provider
directly for clarity.

Epic: none
Release note: None

kyle-a-wong reviewed (approved):
lgtm

xinhaoz commented:
bors r+
